presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1371,6 +1372,11 @@ public SystemSessionProperties(
                         "Use perfectly consistent histories for history based optimizations, even when parts of a query are re-ordered.",
                         featuresConfig.isUsePerfectlyConsistentHistories(),
                         false),
+                integerProperty(
+                        HISTORY_CANONICAL_PLAN_NODE_LIMIT,
+                        "Use history based optimizations only when number of nodes in canonical plan is within this limit. Size of canonical plan can become much larger than original plan in some cases, leading to increased planning time",
```

Comment:
in some cases -> add a note that this is particularly true when there are limit nodes in the plan